python_ta/contracts/__init__.py

```diff
@@ -32,7 +51,13 @@ def new_setattr(self, name, value):
         frame_locals = frame_members['f_locals']
         if self is not frame_locals.get('self'):
             # Only validating if the attribute is not being set in a instance/class method
-            check_invariants(self)
+            init = getattr(wrapped, "__init__")
+            check_invariants(self, rep_invariants, init.__globals__)
+
+    for attr in wrapped.__dict__:
```

Comment:
Right, so in the same vein, it should be possible to wrap the base methods in `type(wrapped).__dict__` rather than `wrapped.__dict__`.